miden-tx/src/compiler/mod.rs

```diff
@@ -96,6 +98,36 @@ impl TransactionCompiler {
         Ok(note_script)
     }
 
+    /// Constructs a [TransactionScript] by compiling the provided source code and checking the
+    /// compatibility of the resulting program with the target account interfaces.
+    pub fn compile_tx_script<T>(
+        &mut self,
+        tx_script_ast: ProgramAst,
+        tx_script_inputs: T,
+        target_account_proc: Vec<ScriptTarget>,
+    ) -> Result<TransactionScript, TransactionCompilerError>
+    where
+        T: IntoIterator<Item = (Word, Vec<Felt>)>,
+    {
+        let (tx_script, code_block) =
+            TransactionScript::new(tx_script_ast, tx_script_inputs, &mut self.assembler).map_err(
+                |e| match e {
+                    TransactionScriptError::ScriptCompilationError(asm_error) => {
+                        TransactionCompilerError::CompileTxScriptFailed(asm_error)
+                    }
+                },
+            )?;
+        for target in target_account_proc.into_iter() {
+            verify_program_account_compatibility(&code_block, &self.get_target_interface(target)?)
+                .map_err(|_| {
+                    TransactionCompilerError::TxScriptIncompatibleWithAccountInterface(
+                        code_block.hash(),
+                    )
+                })?;
+        }
+        Ok(tx_script)
+    }
+
     // TRANSACTION PROGRAM BUILDER
     // --------------------------------------------------------------------------------------------
     /// Compiles a transaction which executes the provided notes and an optional tx script against
```

Comment:
nit: this comment needs to be updated as line 135 does not reflect the current return type.